### PR TITLE
Secure key storage

### PR DESCRIPTION
**Pull request by malroc**
The base idea:

1. We encrypt private keys with AES 256 using 32-byte Argon2 password hash as a key and store only the encrypted version
2. We don't store password hash as the ability to decrypt private key indicates wether we use the correct password or not
3. Any action that requires using private key (basically, that is always creating some type of transaction) requires user to enter password, as the node don't know the unencrypted private key value
4. An exception is made for miner account, for which unencrypted private key is stored in `ETS` (i.e. in memory, never on disc), as we need to submit mining transactions in the background, without additional user input
5. Storing mining private keys in memory means that the mining stops once the node shuts down and cannot be restored without user entering mining account password; this part (i.e. restoration) is not yet implemented as tracking the state of the mining account is a separate (not implemented currently) feature and a subject to a new PR

The PR requires the corresponding changes on the wallet side, hence not well-tested:
1. `/pending_transactions`: pass `password` field
2. `/deposits`: pass `password`, `mining_password`, and `mining_password_confirmation` fields
@AlimovSV could you please adjust the wallet code and check if the new flow works?

### CODE REVIEW

**Comment by AlimovSV** [comment]
@malroc @macgregordennis 
Regarding wallet changes: We have a use case `Deposit More`. In this case the mining account already exists and passing the `mining_password/mining_password_confirmation` looks redundant. But as I can see the server side requires the `mining_password` in any case.

File: lib/credo_core_node_web/controllers/client_api/v1/deposit_controller.ex [hunk 1]
```diff
@@ -27,6 +36,8 @@ defmodule CredoCoreNodeWeb.ClientApi.V1.DepositController do
         |> Pool.propagate_pending_transaction()
         |> elem(1)
 
+      :ets.insert(:mining_accounts, {mining_account.address, mining_private_key})
```
Comment (macgregordennis) [inline lib/credo_core_node_web/controllers/client_api/v1/deposit_controller.ex]: Let's define a function for inserting the mining account private key.

**Comment by macgregordennis** [comment]
> Regarding wallet changes: We have a use case `Deposit More`. In this case, a mining account already exists and passing the `mining_password/mining_password_confirmation` looks redundant. But as I can see the server side requires the `mining_password` in any case.

@AlimovSV Good catch. Let's change the when clauses to account for this scenario.

**Comment by macgregordennis** [comment]
@malroc Generally looks fantastic.

> 5\. Storing mining private keys in memory means that the mining stops once the node shuts down and cannot be restored without user entering mining account password; this part (i.e. restoration) is not yet implemented as tracking the state of the mining account is a separate (not implemented currently) feature and a subject to a new PR

Let's regard that as a high priority.

**Pull request merged**